h/models/user.py

```diff
@@ -19,7 +19,11 @@
 
 USERNAME_MIN_LENGTH = 3
 USERNAME_MAX_LENGTH = 30
-USERNAME_PATTERN = "(?i)^[A-Z0-9._]+$"
+
+# nb. This pattern is used in Python code, JSON schemas and HTML forms, so it
+# needs to use portable syntax.
+USERNAME_PATTERN = "^[A-Za-z0-9_][A-Za-z0-9._]+[A-Za-z0-9_]$"
```

Comment:
Will this affect the [`user-update` API](https://h.readthedocs.io/en/latest/api-reference/v1/#tag/users/paths/~1users~1{username}/patch), for those users whose existing usernames start or end with a `.`?

Reply:
The user update API does not validate the format of the username that I can see. I have added a note about the change in the comment for `USERNAME_PATTERN` to make future readers aware.